pkg/controller/actions/render/kustomize/action_render_manifests_test.go

```diff
@@ -235,14 +236,15 @@ func TestRenderResourcesWithCacheAction(t *testing.T) {
 			)),
 		))
 
-		newHash := rr.Resources[0].GetAnnotations()[annotations.ComponentHash]
-		if i == 1 {
-			g.Expect(newHash).Should(Equal(hash))
-		}
-		if i == 2 {
-			g.Expect(newHash).ShouldNot(Equal(hash))
-		}
+		rc := testutil.ToFloat64(render.RenderedResourcesTotal)
 
-		hash = newHash
+		switch i {
+		case 0:
+			g.Expect(rc).Should(BeNumerically("==", 1))
+		case 1:
+			g.Expect(rc).Should(BeNumerically("==", 2))
+		case 2:
+			g.Expect(rc).Should(BeNumerically("==", 2))
+		}
```

Comment:
not sure i fully understand this case:
so we have 3 replica of dashboard, 
for the 1st one, gauge should set be 1
for the 2nd one, then gauge to 2
and the last one, to 3.
or i mistunderstood this

Reply:
no, this is not about replicas, this is about testing the cache so:
- 1st iteration, the counter should be 1, because it is well nothing did exist before
- 2nd iteration, it should be 2, because the live resource generation has changed
- 3rd iteration, it should still 3, because nothing has changed so no rendering should have been performed and values are taken from the cache

Reply:
ok i see, so the test is to generate a 3 times-loop to minic : 
i = 0, first time creation, no cache, increase in guage
i= 1, render when generateion already as 1, get from cache, increase in gauge
i= 2, render but with same generation value, no increase in gauge
no need more, till be same as i=2

Reply:
So, `case 1,2:` that you do not spend effort to compare symbol by symbol equal branches?

Reply:
in 1,2 is where the action return the previously rendered resources because the source of customization has not changed